geode-core/src/main/java/org/apache/geode/distributed/internal/membership/gms/GMSMembership.java

```diff
@@ -1028,7 +984,11 @@ protected void handleOrDeferViewEvent(MembershipView<ID> viewArg) {
           (MembershipView<InternalDistributedMember>) viewArg,
           (GMSMembership<InternalDistributedMember>) GMSMembership.this);
 
-      messageListener.messageReceived((Message<ID>) v);
+      try {
+        messageListener.messageReceived((Message<ID>) v);
+      } catch (MemberShunnedException e) {
+        logger.error("View installation was blocked by a MemberShunnedException", e);
```

Comment:
This seems like different behavior than before - it would have propagated the exception before. Is that intentional?

Reply:
There was no handler for this exception before - it would have been logged as an uncaught exception.  Making the exception a "checked" exception exposed this problem.

Reply:
Got it